File: java/src/test/java/com/iteso/nintendo/JigglypuffTest.java
package com.iteso.nintendo;

import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.assertEquals;

/**
 * Created by rvillalobos on 2/24/18.
 */
public class JigglypuffTest {
    /**
     * Jigglypuff object.
     */
    Jigglypuff jigglypuff;
    /**
     * Set up for the character.
     */
    @Before
    public void setUp(){
        jigglypuff = new Jigglypuff();
    }
    /**
     * Test of the mainAttack;
     */
    @Test
    public void mainAttack(){
        assertEquals("Attacking opponent with Quick Attack causing damage of 30", pikachu.mainAttack());
    }
    /**
     * Test of the secondAttack;
     */
    @Test
    public void secondAttack(){
        assertEquals("Attacking opponent with Impact Trueno causing damage of 20", pikachu.mainAttack());
    }
}

File: csharp/Pokemon/Pokemon/tests/JigglypuffTest.cs
using NUnit.Framework;
using Pokemon.main;

namespace Pokemon.tests
{
    [TestFixture]
    internal class JigglypuffTest
    {
        public Jigglypuff Jigglypuff;
        [SetUp]
        public void Setup()
        {
            Jigglypuff = new Jigglypuff();
        }

        [Test]
        public void mainAttack()
        {
            Assert.AreEqual("Attacking opponent with Quick Attack causing damage of 30", Jigglypuff.mainAttack());
        }

        [Test]
        public void secondAttack()
        {
            Assert.AreEqual("Attacking opponent with Quick Attack causing damage of 30", Jigglypuff.secondAttack());
        }
    }
}